File: ainpc-core-plugin/src/main/java/ro/ainpc/gui/GuiKey.java
package ro.ainpc.gui;

import java.util.Locale;
import java.util.Optional;

public enum GuiKey {
    MAIN("main", "Hub AINPC"),
    QUEST("quest", "Progresii"),
    QUEST_DETAIL("quest_detail", "Detalii progresie"),
    STORY("story", "Story"),
    WORLD("world", "World"),
    STATS("stats", "Statistici"),
    INTERACT("interact", "Interactiune NPC"),
    ROUTINE("routine", "Rutine NPC"),
    SHOP("shop", "Shop NPC"),
    MANAGER("manager", "Manager NPC"),
    AUDIT("audit", "Audit"),
    DEBUG("debug", "Debug"),
    CONFIRM("confirm", "Confirmare");

    private final String id;
    private final String displayName;

    GuiKey(String id, String displayName) {
        this.id = id;
        this.displayName = displayName;
    }

    public String id() {
        return id;
    }

    public String displayName() {
        return displayName;
    }

    public static Optional<GuiKey> fromId(String rawValue) {
        if (rawValue == null || rawValue.isBlank()) {
            return Optional.of(MAIN);
        }

        String normalized = rawValue.trim().toLowerCase(Locale.ROOT).replace('-', '_');
        normalized = switch (normalized) {
            case "hub", "home", "principal" -> "main";
            case "quests", "questuri", "progression", "progressions", "progresii", "progresie", "log" -> "quest";
            case "questdetail", "quest_details", "quest_detalii", "detalii_quest",
                 "progression_detail", "progression_details", "progresie_detalii", "detalii_progresie" -> "quest_detail";
            case "poveste", "story_state", "story_context", "narativ" -> "story";
            case "map", "lume" -> "world";
            case "stat", "statistics", "statistici" -> "stats";
            case "npc", "interaction", "interactiune", "nearest" -> "interact";
            case "routines", "rutine", "program", "schedule" -> "routine";
            case "admin", "npc_manager", "manager_npc" -> "manager";
            case "debugdump", "dump" -> "debug";
            default -> normalized;
        };

        for (GuiKey key : values()) {
            if (key.id.equals(normalized) || key.name().equalsIgnoreCase(normalized)) {
                return Optional.of(key);
            }
        }
        return Optional.empty();
    }
}

File: ainpc-core-plugin/src/main/kotlin/ro/ainpc/gui/GuiKey.kt
package ro.ainpc.gui

import java.util.Locale
import java.util.Optional

enum class GuiKey(
    private val idValue: String,
    private val displayNameValue: String
) {
    MAIN("main", "Hub AINPC"),
    QUEST("quest", "Progresii"),
    QUEST_DETAIL("quest_detail", "Detalii progresie"),
    STORY("story", "Story"),
    AUTHORING("authoring", "Authoring"),
    WORLD("world", "World"),
    PLACE("place", "Place"),
    REGION("region", "Regiune"),
    STATS("stats", "Statistici"),
    INTERACT("interact", "Interactiune NPC"),
    ROUTINE("routine", "Rutine NPC"),
    SHOP("shop", "Shop NPC"),
    MANAGER("manager", "Manager NPC"),
    AUDIT("audit", "Audit"),
    DEBUG("debug", "Debug"),
    ADMIN_MAPPING("admin_mapping", "Admin Mapping"),
    ADMIN_QUEST("admin_quest", "Admin Quest"),
    QUEST_MAP("quest_map", "Quest Mapping"),
    CONFIRM("confirm", "Confirmare");

    fun id(): String = idValue

    fun displayName(): String = displayNameValue

    companion object {
        @JvmStatic
        fun fromId(rawValue: String?): Optional<GuiKey> {
            if (rawValue.isNullOrBlank()) {
                return Optional.of(MAIN)
            }

            val trimmed = rawValue.trim()
                .lowercase(Locale.ROOT)
                .replace('-', '_')
            val normalized = when (trimmed) {
                "hub", "home", "principal" -> "main"
                "quests", "questuri", "progression", "progressions", "progresii", "progresie", "log" -> "quest"
                "questdetail", "quest_details", "quest_detalii", "detalii_quest",
                "progression_detail", "progression_details", "progresie_detalii", "detalii_progresie" -> "quest_detail"
                "poveste", "story_state", "story_context", "narativ" -> "story"
                "authoring", "quest_authoring", "progress_authoring", "authoring_view" -> "authoring"
                "map", "lume" -> "world"
                "place_detail", "world_place", "loc", "place_details" -> "place"
                "region_detail", "world_region", "regiune" -> "region"
                "stat", "statistics", "statistici" -> "stats"
                "npc", "interaction", "interactiune", "nearest" -> "interact"
                "routines", "rutine", "program", "schedule" -> "routine"
                "admin", "npc_manager", "manager_npc" -> "manager"
                "debugdump", "dump" -> "debug"
                "adminmapping", "admin_mapping", "mapping_admin", "edit_mapping" -> "admin_mapping"
                "adminquest", "admin_quest", "quest_admin", "edit_quest" -> "admin_quest"
                "questmap", "quest_map", "mapping", "create_quest", "edit_quest_map" -> "quest_map"
                else -> trimmed
            }

            for (key in values()) {
                if (key.idValue == normalized || key.name.equals(normalized, ignoreCase = true)) {
                    return Optional.of(key)
                }
            }
            return Optional.empty()
        }
    }
}
